src/mbgl/layermanager/layer_factory.cpp

```diff
@@ -21,6 +25,16 @@ optional<std::string> LayerFactory::getSource(const style::conversion::Convertib
     return source;
 }
 
+std::unique_ptr<Bucket> LayerFactory::createBucket(const BucketParameters&, const std::vector<const RenderLayer*>&) noexcept {
+    assert(false);
```

Comment:
Why not pure virtual?

Reply:
Did not want to repeat the same empty definitions for the concrete factories that do not use this method.